apollo-router/src/trace.rs

```diff
@@ -62,7 +62,16 @@ pub(crate) fn try_initialize_subscriber(
             let provider = builder.with_span_processor(batch).build();
 
             let tracer = provider.tracer("opentelemetry-jaeger", Some(env!("CARGO_PKG_VERSION")));
-            let _ = opentelemetry::global::set_tracer_provider(provider);
+
+            // The call to set_tracer_provider() manipulate a sync RwLock.
+            // Even though this code is sync, it is called from within an
+            // async context. If we don't do this in a separate thread,
+            // it will cause issues with the async runtime that prevents
+            // the router from working correctly.
```

Comment:
:yarn:  The comment should actually mention the ticket ID, a link and summarize what the issue is exactly.

Reply:
Less of "prevents the router from working correctly"
More like "makes the async runtime hang for some reason"

Reply:
resolved in: bef9051cd1767f10423476cea1517dd68e5a7c09